src/riak_cs_wm_object_upload_part.erl

```diff
@@ -202,82 +202,83 @@ parse_body(Body0) ->
             bad
     end.
 
--spec accept_body(#wm_reqdata{}, #context{}) -> {{halt, integer()}, #wm_reqdata{}, #context{}}.
-accept_body(RD, Ctx0=#context{local_context=LocalCtx0,
-                              response_module=ResponseMod,
-                              riak_client=RcPid}) ->
-    #key_context{bucket=DstBucket,
-                 key=Key,
-                 size=Size,
-                 get_fsm_pid=GetFsmPid} = LocalCtx0,
-    catch riak_cs_get_fsm:stop(GetFsmPid),
-    BlockSize = riak_cs_lfs_utils:block_size(),
-    Caller = riak_cs_user:to_3tuple(Ctx0#context.user),
+-spec accept_body(#wm_reqdata{}, #context{}) ->
+                         {{halt, integer()}, #wm_reqdata{}, #context{}}.
+accept_body(RD, #context{local_context=LocalCtx0} = Ctx0) ->
+    catch riak_cs_get_fsm:stop(LocalCtx0#key_context.get_fsm_pid),
 
-    DstKey = list_to_binary(Key),
-
-    {SrcManifest, ExactSize, ReadRcPid} =
-        %% checking existence of "x-amz-copy-source"
-        case riak_cs_copy_object:get_copy_source(RD) of
-            undefined ->
-                %% normal upload, use size claimed via HTTP request
-                {undefined, Size, undefined};
-            {SrcBucket0, SrcKey0} ->
-                %% case copy, use size from copy source manifest
-                {ok, ReadRcPid0} = riak_cs_riak_client:checkout(),
-                {ok, SrcManifest0} = riak_cs_manifest:fetch(ReadRcPid0, SrcBucket0, SrcKey0),
-                {Start,End} = riak_cs_copy_object:copy_range(RD, SrcManifest0),
-                {SrcManifest0, End - Start + 1, ReadRcPid0}
-        end,
-
-    case ExactSize =< riak_cs_lfs_utils:max_content_len() of
-        false ->
-            ResponseMod:api_error(entity_too_large, RD, Ctx0);
+    try
+        {t, {ok, UploadId}} =
+            {t, riak_cs_utils:safe_base64url_decode(
+                  re:replace(wrq:path(RD), ".*/uploads/", "", [{return, binary}]))},
+        {t, {ok, PartNumber}} =
+            {t, riak_cs_utils:safe_list_to_integer(wrq:get_qs_value("partNumber", RD))},
```

Comment:
Sorry, I have to be serious :)

`{t,_}` is matched in `catch` clause right below to respond 400. There might be other error patterns because there was `{t3,_}` at https://github.com/basho/riak_cs/pull/1214/files#diff-d617e0da6709d2bb3eabf098f392f035L277. 
Anyway, these diff lines are just by refactoring and not essential for this PR/commit.